contracts/FIFOMarket.sol

```diff
@@ -147,27 +148,41 @@ contract FIFOMarket is
         suppliers[i] = supplier;
         remainingAmountToFill -= removalAmount;
       }
-
+      numberOfRemovals++;
       if (remainingAmountToFill == 0) {
         break;
       }
     }
 
+    uint256[] memory batchedIds = new uint256[](numberOfRemovals);
+    uint256[] memory batchedAmounts = new uint256[](numberOfRemovals);
+
+    for (
+      uint256 i = _queueHeadIndex;
+      i < _queueHeadIndex + numberOfRemovals;
```

Comment:
We actually shouldn't need to iterate over the whole list again, assuming _queueHeadIndex is correct, we can just iterate over a slice of the list indicated by `numberOfRemovals`